src/test/java/com/vinyl/UpdateProductTest.java

```diff
@@ -0,0 +1,95 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Product;
+import com.vinyl.repository.ProductRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class UpdateProductTest extends ManagerBaseIntegration {
+
+    @Autowired
+    protected TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    protected ProductRepository productRepository;
+
+    protected Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+    }
+
+    @Test
+    public void testWhenOK() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Assertions.assertThat(productRepository.findAll()).doesNotContain(product);
+    }
+
+    @Test
+    public void testWhenProductNameIsInvalid() {
+        request.setProductName("");
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenProductPriceIsInvalid() {
+        request.setPrice(-1.2);
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenProductStockIsInvalid() {
+        request.setStock(0);
```

Comment:
I think u should be able to have a 0 stock of a specific product

Reply:
fixed